### PR TITLE
Add block value to getPayload result

### PR DESCRIPTION
**Pull request by deffrian**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
- Implementation of https://github.com/ethereum/execution-apis/pull/314

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ x ] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ x ] Yes
- [ ] No

### CODE REVIEW

**Comment by rubo** [comment]
@MarekM25 This gonna conflict hard with #4731

**Comment by MarekM25** [comment]
> @MarekM25 This gonna conflict hard with #4731

Yes, could you propose the solution?

**Comment by deffrian** [comment]
> 

I can merge this pr to #4731 after merging it into master

File: src/Nethermind/Nethermind.Merge.Plugin/Data/V2/GetPayloadV2Result.cs
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/V2/GetPayloadV2Result.cs:35]: maybe add ToString to ExecutionPayloadV1 and build on that?

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/V2/GetPayloadV2Handler.cs
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/V2/GetPayloadV2Handler.cs:54]: All this code should be abstracted from V1 and V2 handler to some reusable thing.

**Review by LukaszRozmej — APPROVED** [review]

**Comment by rubo** [comment]
> I can merge this pr to https://github.com/NethermindEth/nethermind/pull/4731 after merging it into master

With this said, I'm making this draft to avoid accidental merges.

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/V2/GetPayloadV2Handler.cs
Comment (deffrian) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/V2/GetPayloadV2Handler.cs:45]: yes, tested manually + unit test

File: src/Nethermind/Nethermind.Evm/Tracing/BlockReceiptsTracer.cs
Comment (deffrian) [reply src/Nethermind/Nethermind.Evm/Tracing/BlockReceiptsTracer.cs:204]: made unit tests

File: src/Nethermind/Nethermind.Core/BlockHeader.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Core/BlockHeader.cs:87]: Do we really need to add these fields here? It looks like that it should be in tracer

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Core/BlockHeader.cs:87]: I do agree, but then we need to move it from tracer to handler, do you have better idea to do that?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Core/BlockHeader.cs:87]: Maybe we can do it better, foundation here: https://github.com/NethermindEth/nethermind/tree/feature/fees_in_getPayload-refactor

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:433]: This logic seems to be incorrect when Eip1559FeeCollector is null.
If we don't have Eip1559FeeCollector burntFees should be still reported. For mainnet we don't have Eip1559FeeCollector.

File: src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs:38]: CurrentBestBlock was better name. Why did we change it?

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs
Comment (deffrian) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:433]: they are reported few lines lower

File: src/Nethermind/Nethermind.Evm/Tracing/CancellationTxTracer.cs
Comment (MarekM25) [reply src/Nethermind/Nethermind.Evm/Tracing/CancellationTxTracer.cs:431]: @deffrian pls cover all new code with unit tests :)

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs
Comment (MarekM25) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:433]: Do we have tests with Eip1559FeeCollector and without Eip1559FeeCollector?

File: src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs
Comment (deffrian) [reply src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs:38]: idk @LukaszRozmej

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:451]: Shouldn't we add IsTracingFees instead of IsTracingReceipt?

**Review by MarekM25 — COMMENTED** [review]
Added a few comments and questions

File: src/Nethermind/Nethermind.Evm/Tracing/FeesTracer.cs
Comment (deffrian) [reply src/Nethermind/Nethermind.Evm/Tracing/FeesTracer.cs:29]: Would be useful I think, for stats or something like that

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs
Comment (MarekM25) [inline src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:433]: In case of freeTransaction it will report fees incorrectly

File: src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/Boost/BoostBlockImprovementContext.cs:38]: if no good reasons please revert it

**Review by MarekM25 — COMMENTED** [review]
One comment about free transactions. 

QQ: How was the PR tested? (besides unit tests, of course :) )

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**
